### ISSUE TITLE
Assigning a new value to a variable with a `Literal` type annotation causes assignment errors.

### ISSUE THREAD

**Issue by HongchengFan** [python/mypy#16104]
**Description**

When I annotate a variable using the `Literal` library and then attempt to reassign the variable, mypy detects assignment errors.

Here is a minimal code snippet to reproduce the crash:
```
from typing import Literal

QUERY: Literal['query'] = 'query'
QUERY = 'queryquery'
QUERY = 'cookie'
```

Print the code in the file.py and execute `mypy file.py`.

**The Actual Output**

```
file.py:4: error: Incompatible types in assignment (expression has type "Literal['queryquery']", variable has type "Literal['query']")  [assignment]
file.py:5: error: Incompatible types in assignment (expression has type "Literal['cookie']", variable has type "Literal['query']")  [assignment]
Found 2 errors in 1 file (checked 1 source file)
```

**The Expected Output**

Code execution won't be blocked, and type-checking won't produce errors, as Python runs it fine.

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used:1.5.1
- Mypy command-line flags: mypy file.py --show-traceback
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.9.0
- Operating system and version: Ubuntu 20.04.6 LTS

**Comment by JelleZijlstra** [comment]
That error is correct. You declared the variable as being of type `Literal["query"]`, so mypy will flag if you assign something of the wrong type to it.
